src/modules/claim/claim.service.ts

```diff
@@ -258,28 +243,35 @@ export class ClaimService {
     filters?: QueryFilters;
     currentUser?: string;
   }) {
+    const rolesOfIssuer = (await this.getBySubject({ subject: issuer, filters: { isAccepted: true } }))
+      .map((r) => r.claimType);
+    const rolesByIssuer = (await this.roleService.getAll())
+      .filter(
+        (r) => r.definition.issuer.did.includes(issuer) || rolesOfIssuer.includes(r.namespace)
+      );
+    this.logger.debug(rolesByIssuer, `roles issuable by ${issuer}`);
+
     const qb = this.claimRepository
-      .createQueryBuilder()
-      .where(':issuer = ANY ("claimIssuer")', { issuer });
+      .createQueryBuilder("claim")
+      .where(':issuer = ANY (claim.claimIssuer)', { issuer });
+    if (rolesByIssuer.length > 0) {
+      qb.orWhere('claim.claimType = IN (:...rolesByIssuer)', { rolesByIssuer });
+    }
 
     if (isAccepted !== undefined) {
-      qb.andWhere('"isAccepted" = :isAccepted', {
+      qb.andWhere('claim.isAccepted = :isAccepted', {
         isAccepted,
       });
     }
 
     if (namespace) {
-      qb.andWhere('"namespace" = :namespace', {
+      qb.andWhere('claim.namespace = :namespace', {
```

Comment:
Isn't of filtering here, maybe could filter at the point of `roleService.getAll()`. Seems a little silly to process all the roles looking for ones this DID can issue when we only care about one namespace. Would be faster to only check if the DID can issue that namespace/role.

Reply:
I think that this flag is intended to filter roles allowed to issue, not roles of issuer